backend/service-user/src/main/java/g55/cs3219/backend/userService/service/AuthenticationService.java

```diff
@@ -112,6 +122,31 @@ public void sendVerificationEmail(User user) {
 
     }
 
+    public User updateUser(Long userId, Map<String, Object> updates, User currentUser) {
+        User userToUpdate = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("User not found"));
+
+        if (!currentUser.isAdmin() && !userToUpdate.getId().equals(currentUser.getId())) {
+            throw new RuntimeException("Forbidden: You can only update your own information.");
+        }
+
+        if (updates.containsKey("username")) {
```

Comment:
Perhaps can extract out as static variable, prolly in future PRs as well not so urgent